### PR TITLE
Cylindrical geometry operators

### PR DESCRIPTION
**Pull request by rtimms**
# Description
Adds cylindrical finite volume method. 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/geometry/standard_spatial_vars.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
 
 y = pybamm.SpatialVariable("y", domain="current collector", coord_sys="cartesian")
 z = pybamm.SpatialVariable("z", domain="current collector", coord_sys="cartesian")
+r = pybamm.SpatialVariable(
+    "r", domain="current collector", coord_sys="cylindrical polar"
+)
```
Comment (tinosulzer) [inline pybamm/geometry/standard_spatial_vars.py:35]: I wonder if we should be more specific with this one, like `r_macro` maybe?

File: pybamm/spatial_methods/finite_volume.py [hunk 2]
```diff
@@ -177,6 +177,14 @@ def divergence(self, symbol, discretised_symbol, boundary_conditions):
             r_edges = pybamm.Vector(r_edges_numpy)
 
             out = divergence_matrix @ ((r_edges ** 2) * discretised_symbol)
+        elif submesh.coord_sys == "cylindrical polar":
+            second_dim_repeats = self._get_auxiliary_domain_repeats(symbol.domains)
+
+            # create np.array of repeated submesh.edges
+            r_edges_numpy = np.kron(np.ones(second_dim_repeats), submesh.edges)
+            r_edges = pybamm.Vector(r_edges_numpy)
```
Comment (tinosulzer) [inline pybamm/spatial_methods/finite_volume.py:185]: reduce code duplication here

File: tests/unit/test_spatial_methods/test_finite_volume/test_finite_volume.py [hunk 3]
```diff
@@ -736,6 +754,36 @@ def test_definite_integral(self):
...
+            2 * np.pi / 3 * (1 - r0 ** 3),
+            decimal=4,
+        )
+        one_over_y = 1 / pts
+        np.testing.assert_array_almost_equal(
+            integral_eqn_disc.evaluate(None, one_over_y),
+            2 * np.pi * (1 - r0),
+            decimal=3,
+        )
+
```
Comment (tinosulzer) [inline tests/unit/test_spatial_methods/test_finite_volume/test_finite_volume.py:786]: we really need to clean up this test file 😅

**Review by tinosulzer — APPROVED** [review]
Thanks, looks good! Lots of conflicts coming up with #1810 :(

File: tests/unit/test_spatial_methods/test_finite_volume/test_finite_volume.py (hunk 3 above)
Comment (rtimms) [reply tests/unit/test_spatial_methods/test_finite_volume/test_finite_volume.py:786]: Haha tell me about it. My favourite bit was the test that said it was testing div(grad(r^2))=6 but it was actually testing div(grad(6))=0...

There's no rush to merge this, so I'll wait, fix conflicts with #1810 and also have a go at tidying up these tests a bit.

**Pull request merged**